clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -66,7 +65,7 @@ public void track(long msgId) {
    * @return true if the given msgId is already tracked otherwise false.
    */
   public boolean seen(long msgId) {
-    boolean seen = (msgId <= highestContiguousMsgId) || nonContiguousMsgIds.contains(msgId);
+    boolean seen = nonContiguousMsgIds.contains(msgId) || msgId <= highestContiguousMsgId;
```

Comment:
Unfortunately, 99.99% of the cases msgId will be greater than highestContigousMsgId.. So that guard is really not helping.  as track() is never called on the active... the hope is contains() is almost always on an empty set except at the beginning of a failover..

Also rather than fixing this at this level, I think we should have the isDuplicate() check itself optimized such that on an active (before failover), map lookups are removed.. Today, it will try a map lookup on a concurrent hash map as follows:

   if (clientUUIDMessageTrackerMap.get(clientId) == null) {
      return false;
    }

After the first failover from a passive to active, the map lookup will find an entry.. and in those cases isDuplicate() check comes to this part of the code, but with msgId almost always > than highestContiguousMsgId. But after a while (post failover) nonContiguousMsgId should always be an emptySet.